Mathlib/Geometry/Manifold/Metrizable.lean

```diff
@@ -19,11 +19,13 @@ open TopologicalSpace
 
 /-- A σ-compact Hausdorff topological manifold over a finite dimensional real vector space is
 metrizable. -/
-theorem ManifoldWithCorners.metrizableSpace {E : Type*} [NormedAddCommGroup E] [NormedSpace ℝ E]
+theorem ModelWithCorners.metrizableSpace {E : Type*} [NormedAddCommGroup E] [NormedSpace ℝ E]
```

Comment:
I find the new name confusing: the *manifold* is metrizable, not the model. `SmoothManifoldWithCorners.metrizableSpace` would be acceptable with me --- but then, the point is that this lemma does not require smoothness (the doc-string even says "topological manifold"). Overall, I'm inclined to let this be. What do you think?

Reply:
I see your point, but I'm not sure `ManifoldWithCorners` explains the theorem statement, either. It is also strange to have one and exactly one item under the namespace `ManifoldWithCorners`, when it isn't a thing on its own.

The way mathlib defines manifolds (with corners) is already not straightforward. I worry that it would be confusing for somebody looking for the manifold API to first stumble on `ManifoldWithCorners` thinking that it is a good starting point.

A descriptive name might be `TopologicalSpace.metrizableSpace_of_finiteDimensional_sigmaCompact_t2`, following `[TopologicalSpace.metrizableSpace_of_t3_second_countable](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Topology/Metrizable/Urysohn.html#TopologicalSpace.metrizableSpace_of_t3_second_countable)`.

Reply:
The instance you quoted is misnamed; filed #18911 for fixing it. Review welcome :-)

Reply:
This instance is about manifolds, though (and the instance for charted spaces already exists --- this is merely a convenience shortcut/serves to point readers to this fact).

Reply:
Ok, I think `Manifold.metrizableSpace` is also fine, following [`Manifold.locallyCompact_of_finiteDimensional`](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Geometry/Manifold/SmoothManifoldWithCorners.html#Manifold.locallyCompact_of_finiteDimensional). I'm also fine with the latter being renamed `ManifoldWithCorners.locallyCompact_of_finiteDimensional` for consistency.

Reply:
Okay then: both seem fine enough for me; can you make that rename? (Or, if you prefer another opinion: ask a maintainer for a preference between those two?)

Reply:
I'll choose `Manifold.` for two reasons.
* `Manifold.locallyCompact_of_finiteDimensional` came first in 2019 (metrizable statement came in 2022).
* There is an existing meaningful namespace called `Manifold`, which contains notations for the API based on `ModelWithCorners`.